File: oop/Java/lv2/zadatak1/Person.java
package zadatak1;

public abstract class Person {
	public static String DEFAULT_NAME = "John Doe";
	public static String DEFAULT_COUNRTY = "Noland";
	public static int DEFAULT_EMOTION = 50;
	final private String name;

	final private String country;

	private int emotion = DEFAULT_EMOTION;

	public Person() {
		this.name = DEFAULT_NAME;
		this.country = DEFAULT_COUNRTY;
	}
	
	public Person(String name, String country, int emotion) {
		if(name == null || country == null || (emotion < 0 || emotion > 100)) {
			throw new IllegalArgumentException("nist ti nevalja buraz");
		}
		this.name = name;
		this.country = country;
		this.emotion = emotion;
	}

	@Override
	public boolean equals(Object o) {
		if (o == this) { 
            return true; 
        } 
		
		if (!(o instanceof Person)) { 
            return false; 
        }
		
		Person p = (Person) o; 
        
        return name.contentEquals(p.name) == true && country.contentEquals(p.country) == true;
	}

	public String getCountry() {
		return country;
	}
	
	public int getEmotion() {
		return emotion;
	}

	public void setEmotion(int emotion) {
		this.emotion = emotion;
	}
	
	public String getName() {
		return name;
	}	
}

File: oop/Kotlin/lv2/zadatak1/Person.kt
package zadatak1


abstract class Person(val name: String = "John Doe", val country: String = "Noland", var emotion: Int = 50) {

    init {
        if (emotion < 0 || emotion > 100) {
            throw Exception("Emotion is out of range! 0 - 100")
        }
    }

    override fun equals(other: Any?): Boolean {
        if (other === this) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as Person?
        return name == that!!.name && country == that!!.country
    }
}